### PR TITLE
Fix visibility macro names when used by a different component (Windows)

### PR DESCRIPTION
**Pull request by j-rivero**
The PR solves a critical problem: when building a new component (new shared object) the visibility keyword can not be the same than a library it links to. This translated means: our systems and components can not use `IGNITION_GAZEBO_VISIBLE` but the custom name created by ign-cmake in the form of `IGNITION_GAZEBO_${COMPONENT_NAME}_VISIBLE`. Otherwise there is a nice list of fanky linker errors all around.

### CODE REVIEW

File: src/systems/breadcrumbs/Breadcrumbs.hh [hunk 1]
```diff
@@ -17,6 +17,7 @@
 #ifndef IGNITION_GAZEBO_SYSTEMS_BREADCRUMBS_HH_
 #define IGNITION_GAZEBO_SYSTEMS_BREADCRUMBS_HH_
 
+#include <optional>
```
Comment (chapulina) [inline src/systems/breadcrumbs/Breadcrumbs.hh:20]: Nit: alphabetize

**Review by chapulina — COMMENTED** [review]
Wow I didn't know those macros even existed! I'm still not sure if the plugins need the visibility to be set though.

Codechecker is sad - otherwise the PR looks good to me.

```
  /github/workspace/include/ignition/gazebo/gui/Gui.hh:63:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
  /github/workspace/src/systems/camera_video_recorder/CameraVideoRecorder.hh:44:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
  /github/workspace/src/systems/joint_position_controller/JointPositionController.hh:75:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
  /github/workspace/src/systems/logical_audio_sensor_plugin/LogicalAudioSensorPlugin.hh:132:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
  /github/workspace/src/systems/multicopter_control/MulticopterVelocityControl.hh:149:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
  /github/workspace/src/systems/multicopter_motor_model/MulticopterMotorModel.hh:37:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
  /github/workspace/src/systems/optical_tactile_plugin/OpticalTactilePlugin.hh:69:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
  /github/workspace/src/systems/triggered_publisher/TriggeredPublisher.hh:157:  Lines should be <= 80 characters long  [whitespace/line_length] [2]
  Total errors found: 8
```

**Comment by azeey** [comment]
> I'm still not sure if the plugins need the visibility to be set though.

Yeah, I don't think we want them to be visible. I would in fact vote for making them `final` classes.

File: src/systems/breadcrumbs/Breadcrumbs.hh (hunk 1 above)
Comment (j-rivero) [reply src/systems/breadcrumbs/Breadcrumbs.hh]: ea7502a

**Comment by j-rivero** [comment]
> I'm still not sure if the plugins need the visibility to be set though.

I've removed visibility from all plugins. 

> Yeah, I don't think we want them to be visible. I would in fact vote for making them `final` classes.

I agree. I would leave the change of moving them to final classes in a different PR.

**Review by chapulina — APPROVED** [review]

**Pull request merged**